File: DataStruct/LeetCode/Javalang/加一_66.java
package leetcode;

public class 加一_66 {
}

class plusOne {
    public int[] plusOne(int[] digits) {
        int len = digits.length;
        if (digits[len - 1] != 9) {
            digits[len - 1]++;
            return digits;
        } else {
            int maxNine = 0;
            for (int i = len - 1; i >= 0; i--) {
                if (digits[i] != 9) {
                    break;
                }
                maxNine++;
            }
            if (maxNine == len) {
                int[] arr = new int[len + 1];
                arr[0] = 1;
                for (int i = 1; i <= len; i++) {
                    arr[i] = 0;
                }
                return arr;
            } else {
                digits[len - maxNine - 1]++;
                for (int i = len - maxNine; i < len; i++) {
                    digits[i] = 0;
                }
                return digits;
            }
        }
    }
}

File: DataStruct/LeetCode/Pylang/加一_66.py
from typing import List

"""
:Author:    iWorld
:Create:    2022/2/22 2:15
:Abstract:  
"""


class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        i = len(digits) - 1
        # [1, 3, 9, 9]
        if (digits[i] != 9):
            digits[i] += 1
            return digits
        else:
            digits[i] = 0
            i -= 1

            while (i >= 0):
                if (digits[i] == 9):
                    digits[i] = 0
                    i -= 1
                    continue
                else:
                    digits[i] += 1
                    return digits

            if (i < 0):
                digits.insert(0, 1)
                return digits


s = Solution()
print(s.plusOne([0,0,0]))
